From e803c6f76f93fd802e4d70ff84aa6258e99119a0 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Fri, 15 Jan 2016 14:04:17 +0800
Subject: [PATCH 44/65] package-kernel: update to v4.4-novena-r4

Signed-off-by: Sean Cross <xobs@kosagi.com>
---
 package-kernel.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package-kernel.sh b/package-kernel.sh
index 77e9355..ed9c655 100755
--- a/package-kernel.sh
+++ b/package-kernel.sh
@@ -1,7 +1,7 @@
 #!/bin/sh
 # A script to build the kernel .deb files, as used by Kosagi.
 threads=$(($(grep '^processor' /proc/cpuinfo  | cut -d: -f2 | tail -n1)+1))
-version=4.4-novena-r3
+version=4.4-novena-r4
 
 if git diff-index --name-only HEAD | grep -qv "^scripts/package"; then
 	echo "Repo has local changes.  Stash or commit them."
-- 
2.7.3

